.github/workflows/release-tarballs.yml

```
@@ -0,0 +1,64 @@
+name: Release StateSmith tarballs on v*.*.* tags
+
+on:
+  push:
+    tags:
+      - "cli-v*.*.*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '22'  # stick to 22 for now. 23 has some bumps causing TypeScript tests to fail: https://github.com/nodejs/node/issues/55417
+
+      - run: npm install -g typescript
+          
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: |
+            6.0.x
+            7.0.x
+            8.0.x
+            
+      - name: Install dependencies
+        run: dotnet restore src/StateSmith.sln
+
+      - name: Build
+        run: |
+          staging="/tmp/staging"
+          publish="/tmp/publish"
+          mkdir -p $publish
+          for arch in linux-arm linux-arm64 linux-musl-arm64 linux-musl-x64 linux-x64 osx-arm64 osx-x64 win-arm64 win-x64 win-x86; do
+            extension=""
+            if [[ $arch == "win"* ]]; then
+              extension=".exe"
+            fi
+            for framework in net6.0 net7.0 net8.0; do
+              mkdir -p $staging
+
+              ( 
+                cd src/StateSmith.Cli 
+                dotnet publish -c Release -r $arch -p:PublishSingleFile=true --self-contained --framework $framework -p:EnableCompressionInSingleFile=true /p:DefineConstants="SS_SINGLE_FILE_APPLICATION" 
```

```suggestion
                dotnet publish -c Release -r $arch -p:PublishSingleFile=true --self-contained --framework $framework -p:EnableCompressionInSingleFile=true /p:DefineConstants="SS_SINGLE_FILE_APPLICATION"  -p:Version=$version
```

We'll need to get the version string from the git tag somehow. The version string will usually be like `cli-v0.1.1`, but it may also look like `cli-v0.1.1-test`